pkg/envtest/server.go

```diff
@@ -46,54 +46,66 @@ It's possible to override some defaults, by setting the following environment va
+	envStopTimeout        = "KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT"
+	envAttachOutput       = "KUBEBUILDER_ATTACH_CONTROL_PLANE_OUTPUT"
+	StartTimeout          = 60
+	StopTimeout           = 60
 
 	defaultKubebuilderControlPlaneStartTimeout = 20 * time.Second
 	defaultKubebuilderControlPlaneStopTimeout  = 20 * time.Second
 )
 
-// getBinAssetPath returns a path for binary from the following list of locations,
-// ordered by precedence:
-// 0. KUBEBUILDER_ASSETS
-// 1. Environment.BinaryAssetsDirectory
-// 2. The default path, "/usr/local/kubebuilder/bin"
-func (te *Environment) getBinAssetPath(binary string) string {
-	valueFromEnvVar := os.Getenv(envKubebuilderPath)
-	if valueFromEnvVar != "" {
-		return filepath.Join(valueFromEnvVar, binary)
-	}
+// internal types we expose as part of our public API
 
-	if te.BinaryAssetsDirectory != "" {
-		return filepath.Join(te.BinaryAssetsDirectory, binary)
-	}
+type (
+	// ControlPlane is the re-exported ControlPlane type from the internal integration package
+	ControlPlane = controlplane.ControlPlane
 
-	return filepath.Join(defaultKubebuilderPath, binary)
-}
+	// APIServer is the re-exported APIServer 	from the internal integration package
+	APIServer = controlplane.APIServer
+
+	// Etcd is the re-exported Etcd 	from the internal integration package
+	Etcd = controlplane.Etcd
+
+	// User represents a Kubernetes user to provision for auth purposes.
+	User = controlplane.User
+
+	// AuthenticatedUser represets a Kubernetes user that's been provisioned.
+	AuthenticatedUser = controlplane.AuthenticatedUser
+
+	// ListenAddr indicates the address and port that the API server should listen on.
+	ListenAddr = process.ListenAddr
 
-// ControlPlane is the re-exported ControlPlane type from the internal integration package
-type ControlPlane = integration.ControlPlane
+	// SecureServing contains details describing how the API server should serve
+	// its secure endpoint.
+	SecureServing = controlplane.SecureServing
 
-// APIServer is the re-exported APIServer type from the internal integration package
-type APIServer = integration.APIServer
+	// Authn is an authentication method that can be used with the control plane to
+	// provision users.
+	Authn = controlplane.Authn
 
-// Etcd is the re-exported Etcd type from the internal integration package
-type Etcd = integration.Etcd
+	// Arguments allows configuring a process's flags.
+	Arguments = process.Arguments
+
+	// Arg is a single flag with one or more values.
+	Arg = process.Arg
+)
+
+var (
+	// EmptyArguments constructs a new set of flags with nothing set.
+	//
+	// This is mostly useful for testing helper methods -- you'll want to call
+	// Configure on the APIServer (or etcd) to configure their arguments.
+	EmptyArguments = process.EmptyArguments()
+)
 
 // Environment creates a Kubernetes test environment that will start / stop the Kubernetes control plane and
 // install extension APIs
 type Environment struct {
```

Comment:
How about adding a method `GetConfigFor(username string, groups ...string)(*rest.Config, error`)? Right now this is somewhat hidden in `Environent.Controlplane.AddUser`

Reply:
:+1: